crates/matrix-sdk/src/room/common.rs

```diff
@@ -317,6 +319,70 @@ impl Common {
         }
     }
 
+    async fn request_encryption_state(&self) -> Result<Option<RoomEncryptionEventContent>> {
+        if let Some(mutex) = self
+            .client
+            .inner
+            .encryption_state_request_locks
+            .get(self.inner.room_id())
+            .map(|m| m.clone())
```

Comment:
This can probably be `.cloned()`, right?

Reply:
No, sadly this does not work, it seems to access some Derefed type or something, that cloned does not.